File: FinalProject_OOP_Java/src/Stall/Stall.java
package Stall;

import java.io.BufferedReader;
import java.io.File;
import java.io.FileNotFoundException;
import java.io.FileReader;
import java.io.IOException;
import java.util.LinkedList;
import java.util.List;

import Device.Device;
import Device.Device_List;
import Door.Door;
import Door.Door_List;
import Guard.Guard;
import Guard.Guard_List;
import Vehicle.Vehicle;

public class Stall {
	 // fields
    private String Stall_sID;
    private String Stall_sName;
    private int Stall_iNumCar;
    private int Stall_iNumBike;
    private int Stall_iNumElec;
    private List<Door> Stall_lList_door;
    private List<Device> Stall_lList_device;
    private List<Guard> Stall_lList_guard;
    private List<Vehicle> Stall_lList_vehicle;
    // Properties
	public String getStall_sID() {
		return Stall_sID;
	}
	public void setStall_sID(String stall_sID) {
		Stall_sID = stall_sID;
	}
	public String getStall_sName() {
		return Stall_sName;
	}
	public void setStall_sName(String stall_sName) {
		Stall_sName = stall_sName;
	}
	public int getStall_iNumCar() {
		return Stall_iNumCar;
	}
	public void setStall_iNumCar(int stall_iNumCar) {
		Stall_iNumCar = stall_iNumCar;
	}
	public int getStall_iNumBike() {
		return Stall_iNumBike;
	}
	public void setStall_iNumBike(int stall_iNumBike) {
		Stall_iNumBike = stall_iNumBike;
	}
	public int getStall_iNumElec() {
		return Stall_iNumElec;
	}
	public void setStall_iNumElec(int stall_iNumElec) {
		Stall_iNumElec = stall_iNumElec;
	}
	public List<Door> getStall_lList_door() {
		return Stall_lList_door;
	}
	public void setStall_lList_door(List<Door> stall_lList_door) {
		Stall_lList_door = stall_lList_door;
	}
	public List<Device> getStall_lList_device() {
		return Stall_lList_device;
	}
	public void setStall_lList_device(List<Device> stall_lList_device) {
		Stall_lList_device = stall_lList_device;
	}
	public List<Guard> getStall_lList_guard() {
		return Stall_lList_guard;
	}
	public void setStall_lList_guard(List<Guard> stall_lList_guard) {
		Stall_lList_guard = stall_lList_guard;
	}
	public List<Vehicle> getStall_lList_vehicle() {
		return Stall_lList_vehicle;
	}
	public void setStall_lList_vehicle(List<Vehicle> stall_lList_vehicle) {
		Stall_lList_vehicle = stall_lList_vehicle;
	}
    
	public Stall()
    {
         this.Stall_lList_door = new LinkedList<Door>();
         this.Stall_lList_device = new LinkedList<Device>();
         this.Stall_lList_guard = new LinkedList<Guard>();
         this.Stall_lList_vehicle = new LinkedList<Vehicle>();
    }
	
	public Stall(String Stall_ID,String Stall_name, int Stall_NumCar,int Stall_Numbike,int Stall_Elec) 
    {
        this.Stall_sID = Stall_ID;
        this.Stall_sName = Stall_name;
        this.Stall_iNumCar = Stall_NumCar;
        this.Stall_iNumBike = Stall_Numbike;
        this.Stall_iNumElec = Stall_Elec;
    }
	// Methods
	public void Stall_Docfile(String path1) throws FileNotFoundException
	{
		  File file= new File(path1);
	         try (BufferedReader reader = new BufferedReader(new FileReader(file))) {
				try
				 {
				     String line = reader.readLine();
				     while (line != null)
				     {
				         String[] arr1 = line.split(";");

				         if (arr1.length == 5)
				         {
				        	 this.Stall_sID = arr1[0];
			                 this.Stall_sName = arr1[1];
			                 this.Stall_iNumCar = Integer.parseInt(arr1[2]);
			                 this.Stall_iNumBike = Integer.parseInt(arr1[3]);
			                 this.Stall_iNumElec = Integer.parseInt(arr1[4]);
				         }
				         else throw new Exception("EROLL");
				         line = reader.readLine();
				     }
				     reader.close();
				 }
				 catch (Exception e)
				 {
				     System.out.print(e);
				 }
			} catch (FileNotFoundException e) {
				throw e;
			} catch (IOException e) {
				// TODO Auto-generated catch block
				e.printStackTrace();
			}
	}
	
	public void Stall_Ouput()
    { 
        System.out.println("ID of Stall:"+this.Stall_sID + ",Name of stall:" + this.Stall_sName + ",Maxium number of car:" + this.Stall_iNumCar + ",Maxium number of bike:" + this.Stall_iNumBike + ",Maxium of electric car:" + this.Stall_iNumElec);
        // Output list of device of Stall
        System.out.println("___List of Devices of"+this.Stall_sName+":___");
        Device_List.Ouput(this.Stall_lList_device);
        // Output List of Guard of Stall
        System.out.println("___List of Guards of" + this.Stall_sName + ":___");
        Guard_List.OuputList(this.Stall_lList_guard);
        // Ouput List of Door of Stall
        System.out.println("___List of Doors of" + this.Stall_sName + ":___");
        Door_List.Ouput_1(this.Stall_lList_door);

    }
    
}

File: FinalProject_OOP_C#/Stall.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace FinalProject_OOP
{
    public class Stall
    {
        // fields
        private string Stall_sID;
        private string Stall_sName;
        private int Stall_iNumCar;
        private int Stall_iNumBike;
        private int Stall_iNumElec;
        private List<Door> Stall_lList_door;
        private List<Device> Stall_lList_device;
        private List<Guard> Stall_lList_guard;
        private List<Vehicle> Stall_lList_vehicle;
        // Properties
        public string Stall_ID
        {
            get { return this.Stall_sID; }
            set { this.Stall_sID = value;}
        }
        public string Stall_Name
        {
            get { return this.Stall_sName; }
            set { this.Stall_sName = value;}
        }
        public int Stall_NumCar
        {
            get { return this.Stall_iNumCar; }
            set { this.Stall_iNumCar=value;}
        }
        public int Stall_NumBike
        {
            get { return this.Stall_iNumBike; }
            set { this.Stall_iNumBike = value; }
        }
        public int Stall_NumElec
        {
            get { return this.Stall_iNumElec; }
            set { this.Stall_iNumElec = value; }
        }
        public List<Door> Stall_List_Door
        {
            get { return this.Stall_lList_door; }
            set { this.Stall_lList_door = value;}
        }
        public List<Device> Stall_List_Device
        {
            get { return this.Stall_lList_device; }
            set { this.Stall_lList_device = value; }
        }
        public List<Guard> Stall_List_Guards
        {
            get { return this.Stall_lList_guard; }
            set { this.Stall_lList_guard = value;  }
        }
        public List<Vehicle> Stall_List_Vehicles
        {
            get { return this.Stall_lList_vehicle; }
            set { this.Stall_lList_vehicle = value; }
        }
        // Constructor
        public Stall()
        {
             this.Stall_lList_door = new List<Door>();
             this.Stall_lList_device = new List<Device>();
             this.Stall_lList_guard = new List<Guard>();
             this.Stall_lList_vehicle = new List<Vehicle>();
        }
        public Stall(string Stall_ID,string Stall_name, int Stall_NumCar,int Stall_Numbike,int Stall_Elec) 
        {
            this.Stall_ID = Stall_ID;
            this.Stall_Name = Stall_name;
            this.Stall_NumCar = Stall_NumCar;
            this.Stall_NumBike = Stall_Numbike;
            this.Stall_NumElec = Stall_Elec;
        }
        // Finalizer
        ~Stall()
        { }
        // Methods
        public void Stall_Docfile(string path1)
        { 
            // Stall information
            StreamReader sr1=new StreamReader(path1,Encoding.UTF8);
            string line=sr1.ReadLine(); 
            while(line!=null) 
            {
                string[] arr1 = line.Split(";");
                if(arr1.Length==5)
                {
                    this.Stall_ID = arr1[0];
                    this.Stall_Name = arr1[1];
                    this.Stall_NumCar = int.Parse(arr1[2]);
                    this.Stall_NumBike = int.Parse(arr1[3]);
                    this.Stall_NumElec = int.Parse(arr1[4]);
                }
                line = sr1.ReadLine();
            }
        }
        public void String_to_Stall(string line)
        {

            string[] arr1 = line.Split(";");
            if (arr1.Length == 5)
            {
                this.Stall_ID = arr1[0];
                this.Stall_Name = arr1[1];
                this.Stall_NumCar = int.Parse(arr1[2]);
                this.Stall_NumBike = int.Parse(arr1[3]);
                this.Stall_NumElec = int.Parse(arr1[4]);
            }
        }
        public void Stall_Ouput()
        { 
            Console.WriteLine("ID of Stall:"+this.Stall_ID + ",Name of stall:" + this.Stall_Name + ",Maxium number of car:" + this.Stall_NumCar + ",Maxium number of bike:" + this.Stall_NumBike + ",Maxium of electric car:" + this.Stall_NumElec);
            // Output list of device of Stall
            Console.WriteLine("___List of Devices of"+this.Stall_Name+":___");
            Device_List.OuputList(this.Stall_List_Device);
            // Output List of Guard of Stall
            Console.WriteLine("___List of Guards of" + this.Stall_Name + ":___");
            Guard_List.OuputList(this.Stall_List_Guards);
            // Ouput List of Door of Stall
            Console.WriteLine("___List of Doors of" + this.Stall_Name + ":___");
            Door_List.Ouput_1(this.Stall_List_Door);

        }
    }
}
